sdc/crypto/jwe_helper.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from jwcrypto import jwe
+from jwcrypto.jwe import InvalidJWEData
+
+
+from sdc.crypto.exceptions import InvalidTokenException, MissingKidException, MissingKeyException
+from sdc.crypto.helper import extract_kid_from_header
+
+logger = logging.getLogger(__name__)
+
+
+class JWEHelper:
+    '''
+    Helper methods for encrypting a JWS/JWT. This class assumes you have used the RSA-OAP-A256GCM alogrithm
+    and the required private key is available in the keystore.
+    '''
+    @staticmethod
+    def decrypt(encrypted_token, key_store=None, purpose=None):
+        try:
+            jwe_token = jwe.JWE(algs=['RSA-OAEP', 'A256GCM'])
+            jwe_token.deserialize(encrypted_token)
+
+            jwe_kid = extract_kid_from_header(encrypted_token)
+            if not jwe_kid:
+                raise MissingKidException
+
+            logger.info("Decrypting JWE kid is {}".format(jwe_kid))
+
+            private_jwk = key_store.get_private_key_by_kid(purpose, jwe_kid).as_jwk()
+            if not private_jwk:
+                raise MissingKeyException
+
+            jwe_token.decrypt(private_jwk)
```

Comment:
I did try that, but it fails loads of tests because the tests assume that the text in the exception matches that generated by the underlying library in the call to   jwe_token.deserialize(encrypted_token)

It's a lot of effort to change, but agree with you it would be better that way. I'll add it to the test issue here: https://github.com/ONSdigital/sdc-cryptography/issues/13 and fix it then